coord/src/module.c

```diff
@@ -918,9 +918,13 @@ static void proccessKNNSearchReply(MRReply *arr, searchReducerCtx *rCtx, RedisMo
         // Current result is smaller then upper bound, replace them.
         largest = heap_poll(reduceSpecialCaseCtx->knn.pq);
         heap_offerx(reduceSpecialCaseCtx->knn.pq, resWrapper);
+        free(rCtx->cachedResult);
         rCtx->cachedResult = largest->result;
         rm_free(largest);
       }
+      else {
+        free(res);
```

Comment:
`rm_free`?

Reply:
I thought about it. Those were allocated on line 545 with `malloc`, not `rm_malloc`. I considered changing the allocation to `rm_malloc` and using `rm_free` here, but I didn't know how much else it would affect.

Reply:
@efeldblum-redis We should stick to `rm_alloc`/`rm_free`. All allocators of external libraries were also tweaked to do so.